clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangelogConsumerClientFactory.java

```diff
@@ -73,6 +73,14 @@ public synchronized <K, V> VeniceChangelogConsumer<K, V> getChangelogConsumer(St
     });
   }
 
+  /**
+   * Creates a VeniceChangelogConsumer with consumer id. This is used to create multiple consumers so that
+   * each consumer can only subscribe to certain partitions. Multiple such consumers can work in parallel.
+   */
+  public synchronized <K, V> VeniceChangelogConsumer<K, V> getChangelogConsumer(String storeName, String consumerId) {
+    return getChangelogConsumer(storeName + "-" + consumerId);
```

Comment:
The key change is to "identify" the `consumer` as a many->one mapping to the `store`. Here, `getChangelogConsumer` expects a store name so passing this concatenation might end up breaking code that expects a valid store name. Were you able to validate that the called code has no problem with it?
I suggest inverting the overloading. Have the current impl take an optional `consumerId` argument additionally and add a backwards compatible method that invokes the updated impl with an `Optional.empty` `consumerId`.